control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -4338,11 +4338,32 @@ func (r *HostedControlPlaneReconciler) destroyAWSDefaultSecurityGroup(ctx contex
 		}
 	}
 
+	originalHCP := hcp.DeepCopy()
+	condition := &metav1.Condition{
+		Type: string(hyperv1.AWSDefaultSecurityGroupDeletion),
+	}
+
 	if _, err = r.ec2Client.DeleteSecurityGroupWithContext(ctx, &ec2.DeleteSecurityGroupInput{
 		GroupId: sg.GroupId,
 	}); err != nil {
+		condition.Message = err.Error()
```

Comment:
can you confirm this error returned by aws doesn't contain any variant ID that results in the condition being updated in loop? If so you might want to handcraft the error message removing the variable bit. See e.g what we do in private link controllers.

Also, what if the other calls above within destroyAWSDefaultSecurityGroup fail?
I would expect we set this condition based on the output of destroyAWSDefaultSecurityGroup

Reply:
Yeah, I cannot confirm that. Maybe the best way is just create a custom message, something like: "Error deleting default security group during deprovision".
Humm, makes sense, I think I can create/update the condition in a lvl above.